mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearAlgorithm.scala

```diff
@@ -114,7 +114,7 @@ abstract class StreamingLinearAlgorithm[
     if (model.isEmpty) {
       throw new IllegalArgumentException("Model must be initialized before starting prediction.")
     }
-    data.map(model.get.predict)
```

Comment:
`.map(model.get.predict)` executes `model.get.predict` directly. Check the following:

~~~scala
val rdd = sc.parallelize(0 until 10)
var func = (x: Int) => 1
val rdd1 = rdd.map(func)
val rdd2 = rdd.map(x => func(x))
func = (x: Int) => 2
rdd1.collect()
rdd2.collect()
~~~